pylink/jlink.py

```diff
@@ -4719,3 +4719,148 @@ def swo_read_stimulus(self, port, num_bytes):
+        Args:
+          self (JLink): the ``JLink`` instance
+
+        Returns:
+          The number of configured down buffers on the target.
+
+        Raises:
+          JLinkRTTException if the underlying JLINK_RTTERMINAL_Control call fails.
+        """
+
+        cmd = enums.JLinkRTTCommand.GETNUMBUF
+        dir = ctypes.c_int(enums.JLinkRTTDirection.DOWN)
+        return self.rtt_control(cmd, dir)
+
+    @open_required
+    def rtt_read(self, buffer_index, num_bytes):
+        """Reads data from the RTT buffer.
+
+        This method will read at most num_bytes bytes from the specified
+        RTT buffer. The data is automatically removed from the RTT buffer.
+        If there are not num_bytes bytes waiting in the RTT buffer, the
+        entire contents of the RTT buffer will be read.
+
+        Args:
+          self (JLink): the ``JLink`` instance
+          buffer_index (int): the index of the RTT buffer to read from
+          num_bytes (int): the maximum number of bytes to read
+
+        Returns:
+          A list of bytes read from RTT.
+
+        Raises:
+          JLinkRTTException if the underlying JLINK_RTTERMINAL_Read call fails.
+        """
+
+        buf = (ctypes.c_ubyte * num_bytes)()
+        bytes_read = self._dll.JLINK_RTTERMINAL_Read(buffer_index, buf, num_bytes)
+
+        if bytes_read < 0:
+            raise errors.JLinkRTTException(bytes_read)
+
+        return list(buf)[:bytes_read]
+
+    @open_required
+    def rtt_write(self, buffer_index, data):
+        """Writes data to the RTT buffer.
+
+        This method will write at most len(data) bytes to the specified RTT
+        buffer.
+
+        Args:
+          self (JLink): the ``JLink`` instance
+          buffer_index (int): the index of the RTT buffer to write to
+          data (list): the list of bytes to write to the RTT buffer
+
+        Returns:
+          The number of bytes successfully written to the RTT buffer.
+
+        Raises:
+          JLinkRTTException if the underlying JLINK_RTTERMINAL_Write call fails.
+        """
+
+        buf_size = len(data)
+        buf = (ctypes.c_ubyte * buf_size)(*bytearray(data))
+        bytes_written = self._dll.JLINK_RTTERMINAL_Write(buffer_index, buf, buf_size)
+
+        if bytes_written < 0:
+            raise errors.JLinkRTTException(bytes_written)
+
+        return bytes_written
+
+    @open_required
+    def rtt_control(self, command, config):
+        """Issues an RTT Control command.
+
+        All RTT control is done through a single API call which expects
+        specifically laid-out configuration structures.
+
+        Args:
+          self (JLink): the ``JLink`` instance
+          command (int): the command to issue (see enums.JLinkRTTCommand)
+          config (ctypes type): the configuration to pass by reference.
+
+        Returns:
+          An integer containing the result of the command.
+        """
+
+        config_byref = ctypes.byref(config) if config is not None else None
```

Comment:
This seems fine, but should we implement the command for getting a buffer description?

Reply:
Responded on main comment thread: (yes, but not in this PR)